src/course-search/SearchBox.jsx

```diff
@@ -4,19 +4,25 @@ import classNames from 'classnames';
 import { SearchField } from '@edx/paragon';
 import { connectSearchBox } from 'react-instantsearch-dom';
 
+import { sendTrackEvent } from '@edx/frontend-platform/analytics';
+
 import { deleteRefinementAction, setRefinementAction } from './data/actions';
 import { STYLE_VARIANTS } from '../constants';
 import { SearchContext } from './SearchContext';
 import { QUERY_PARAM_FOR_PAGE, QUERY_PARAM_FOR_SEARCH_QUERY } from './data/constants';
 
 export const searchText = 'Search courses';
+// this prefix will be combined with one of the SearchBox props to create a full tracking event name
+// only if event name prop is provided by user. In the absence of the tracking name prop,
+// no tracking event will be sent.
+export const SEARCH_EVENT_NAME_PREFIX = 'edx.enterprise';
```

Comment:
I think we'll eventually want to add similar tracking to learner portal and admin portal, in which case, this prefix may not be specific enough.

Reply:
For learner/admin portals, couldn't we just change the `trackingName` prop passed to the `SearchData` component to include `learner_portal` and `admin_portal` respectively, resulting in `edx.enterprise.learner_portal.catalog_search` as the event name?

Reply:
yeah that was my vision, is for users to provide that middle portion of the name (then prefix and suffix would then just be added by frontend-enterprise)

Reply:
yes so 

```
<SearchData trackingName="admin_portal">
```
kind of thing is what I am thinking. It just needs to be unique per product I guess.